pkg/vault/contracts/Vault.sol

```diff
@@ -872,6 +873,13 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
                 _totalSupply(params.pool),
                 bptAmountIn
             );
+
+            if (_addLiquidityCalled().tGet(params.pool)) {
```

Comment:
Note: if we clear this flag here, it's easy to bypass.

You add a huge amount, make a decoy proportional exit, and then a big exit.
Whenever the flag is set, it'll be set for the rest of the tx. A more correct approach would be to clean it when `transient` modifier ends, but we would need a transient set for that and I really don't think it's worth the added complexity.

Reply:
Agree that's probably not worth it for an edge case. Although... maybe aggregators might bundle lots of transactions and come across that, since it just checks the pool, not the pool + user, so this could also get triggered on bundled transactions with multiple unrelated users doing liquidity operations on a popular pool. Maybe worth investigating if that's possible? If one person added and another person removed, the second one would get unexpectedly taxed.

Though even storing the user wouldn't be foolproof, as someone could craft a transaction where the add and remove steps were done from separate addresses controlled by the same account.

Reply:
Double checked with @danielmkm and this should not be a problem in general.